modules/402-ingress-nginx/crds/ingress-nginx.yaml

```diff
@@ -610,16 +610,43 @@ spec:
+
+                      If there is no hostname matching the requested host name, the request is handed over to NGINX on the configured passthrough proxy port (default: 442), 
+                      which proxies the request to the default backend.
+                    
+                      > **Caution!** Unlike HTTP backends, traffic to Passthrough backends is sent to the clusterIP of the backing Service instead of individual Endpoints.
+                    
+                      > **Caution!** Because SSL Passthrough works on layer 4 of the OSI model (TCP) and not on the layer 7 (HTTP), using SSL Passthrough invalidates all the other annotations set on an Ingress object.
                     * `HostPort` — Ingress controller is deployed and available through nodes' ports via `hostPort`.
                     * `HostPortWithProxyProtocol` — Ingress controller is deployed and available through nodes' ports via `hostPort, it uses proxy-protocol to get a real IP of the client.
 
                       > **Caution!** Make sure that requests to the Ingress are sent from trusted sources when using this inlet. The `acceptRequestsFrom` parameter can help you with defining trusted sources.
+                    * `HostPortWithSSLPassthrough` - Ingress controller is deployed and available through nodes' ports via `hostPort, it enables SSLPassthrough option and backends can be configured to accept traffic without SSL termination.
+                    
+                      SSL Passthrough leverages SNI and reads the virtual domain from the TLS negotiation, which requires compatible clients. After a connection has been accepted by the TLS listener, 
+                      it is handled by the controller itself and piped back and forth between the backend and the client.
+
+                      If there is no hostname matching the requested host name, the request is handed over to NGINX on the configured passthrough proxy port (default: 442), 
+                      which proxies the request to the default backend.
+                    
+                      Settings are required in [spec.HostPortWithSSLPassthrough](#ingressnginxcontroller-v1-spec-hostportwithsslpassthrough).
+                    
+                      > **Caution!** SSLPassthrough feature is implemented by intercepting all traffic on the configured HTTPS port (default: 443) and handing it over to a local TCP proxy. This bypasses NGINX completely and introduces a non-negligible performance penalty.
+                    
+                      > **Caution!** Unlike HTTP backends, traffic to Passthrough backends is sent to the clusterIP of the backing Service instead of individual Endpoints.
+                    
+                      > **Caution!** Because SSL Passthrough works on layer 4 of the OSI model (TCP) and not on the layer 7 (HTTP), using SSL Passthrough invalidates all the other annotations set on an Ingress object.
```

Comment:
```suggestion
                    * `HostPortWithSSLPassthrough` - Ingress controller is deployed and available through nodes' ports via `hostPort.  It enables SSL Passthrough feature, allowing backends to be configured to accept SSL traffic directly, without terminating it at the Ingress controller.

                      Settings are required in [spec.HostPortWithSSLPassthrough](#ingressnginxcontroller-v1-spec-hostportwithsslpassthrough).

                      SSL Passthrough leverages SNI and reads the virtual domain from the TLS negotiation, which requires compatible clients. After a connection has been accepted by the TLS listener, it is handled by the controller itself and piped back and forth between the backend and the client.

                      If there is no hostname matching the requested host name, the request is handed over to NGINX on the configured passthrough proxy port (default: 442), 
                      which proxies the request to the default backend.

                      > **Caution!** SSL Passthrough feature is implemented by intercepting all traffic on the configured HTTPS port (default: 443) and handing it over to a local TCP proxy. This bypasses NGINX completely and introduces a non-negligible performance penalty.
                      >
                      > **Caution!** Unlike HTTP backends, traffic to SSL Passthrough backends is sent to the ClusterIP of the backing Service instead of individual Endpoints.
                      >
                      > **Caution!** Because SSL Passthrough works on layer 4 of the OSI model (TCP) and not on the layer 7 (HTTP), using SSL Passthrough invalidates all the other annotations set on an Ingress object.
```